src/Compilers/CSharp/Portable/Compilation/InitializerSemanticModel.cs

```diff
@@ -49,7 +49,7 @@ internal static InitializerSemanticModel Create(SyntaxTreeSemanticModel containi
         internal static InitializerSemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, CSharpSyntaxNode syntax, PropertySymbol propertySymbol, Binder rootBinder)
         {
             Debug.Assert(containingSemanticModel != null);
-            Debug.Assert(syntax.IsKind(SyntaxKind.PropertyDeclaration));
+            Debug.Assert(syntax.IsKind(SyntaxKind.PropertyDeclaration) || syntax.IsKind(SyntaxKind.DataPropertyDeclaration));
```

Comment:
💭 Can we call this a `PropertyFieldDeclaration` to align with `EventFieldDeclaration`?